packages/async/src/babel-plugin.ts

```diff
@@ -0,0 +1,143 @@
+import * as Types from '@babel/types';
+import {NodePath, Binding} from '@babel/traverse';
+
+const DEFAULT_PROCESS_PACKAGES = {
+  '@shopify/react-async': ['createAsyncComponent'],
+  '@shopify/react-graphql': ['createAsyncQueryComponent'],
+};
+
+interface State {
+  processPackages?: Map<string, string[]>;
+  opts?: {packages?: {[key: string]: string[]}};
+}
+
+export default function asyncBabelPlugin({types: t}: {types: typeof Types}) {
+  return {
+    visitor: {
+      Program(_path: NodePath<Types.Program>, state: State) {
+        state.processPackages = new Map(
+          Object.entries(
+            (state.opts && state.opts.packages) || DEFAULT_PROCESS_PACKAGES,
+          ),
+        );
+      },
+      ImportDeclaration(path: NodePath<Types.ImportDeclaration>, state: State) {
+        const {processPackages} = state;
+
+        if (!(processPackages instanceof Map)) {
+          return;
+        }
+
+        const source = path.node.source.value;
+
+        const processImports = processPackages.get(source) || [];
+
+        if (processImports.length === 0) {
+          return;
+        }
+
+        const factoryImports = path.get('specifiers').filter(specifier => {
```

Comment:
Sorry, maybe I'm blind, but where does `path` come from here?

Reply:
Path is an extremely common term to use for this in babel plugins, it is what [their handbook](https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/plugin-handbook.md#toc-unit-testing) uses